Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,15 +57,31 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    bool check = true;
+
     QStringList sections = line.split(BATDATA_DELIMITER);
 
     QString timeString = sections.at(0);
+
+    if (sections.size() != 3)
+        return false;
+        //QTextStream(stdout) <<"error ";
+
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    batteryData.voltage = sections.at(1).toDouble(&check);
 
-    batteryData.current = sections.at(2).toDouble();
+    batteryData.current = sections.at(2).toDouble(&check);
```

Comment:
There's a bit of a logical error here. If the voltage has an error, but the current is fine then `check` would still return true... how can we fix that up?